core/dbt/include/global_project/macros/materializations/incremental/incremental.sql

```diff
@@ -7,15 +7,31 @@
   {% set existing_relation = load_relation(this) %}
   {% set tmp_relation = make_temp_relation(this) %}
 
+  {% set on_schema_change = incremental_validate_on_schema_change(config.get('on_schema_change')) %}
+
   {{ run_hooks(pre_hooks, inside_transaction=False) }}
 
   -- `BEGIN` happens here:
   {{ run_hooks(pre_hooks, inside_transaction=True) }}
 
+  {% do run_query(create_table_as(True, tmp_relation, sql)) %}
+  {% set schema_changed = check_for_schema_changes(tmp_relation, target_relation) %}
```

Comment:
Hm hm hm. On second thought, I don't think we want to create the temp table _before_ checking to see if we should full refresh for `should_full_refresh()` or `existing_relation.is_view` reasons. And if we do need to full-refresh for schema-change reasons, we'll want to swap the existing table with a new _permanent_ table, rather than a new _temp_ table.

Let's think more about the right way to structure this logical flow. I bet we can come up with something that makes sense and reads clearly.